pkg/autodiscovery/providers/kubelet.go

```diff
@@ -109,6 +122,17 @@ func parseKubeletPodlist(podlist []*kubelet.Pod) ([]integration.Config, error) {
 
 			configs = append(configs, c...)
 		}
+		adErrors = append(adErrors, utils.ValidateAnnotationsMatching(pod.Metadata.Annotations, containerIdentifiers, containerNames, adPrefix)...)
+		for _, err := range adErrors {
+			log.Errorf("Can't parse template for pod %s: %s", pod.Metadata.Name, err)
+			_, found := ADErrors[pod.Metadata.Name]
+			if !found {
+				ADErrors[pod.Metadata.Name] = map[string]bool{err.Error(): true}
+			} else {
+				ADErrors[pod.Metadata.Name][err.Error()] = true
```

Comment:
not sure I understand why we need the boolean, wouldn't a `map[string]string` be enough?

Reply:
I used `map[string]bool` to try and prevent duplicate errors from showing per pod. Would another data type be better to use for that?

Reply:
seems like the error messages contain the annotation key right? and we know the annotation key is unique per pod so we don't risk a collision